src/modules/producer.js

```diff
@@ -172,7 +172,7 @@ class Producer {
     // default options are persistent and durable because we do not want to miss any outgoing message
     // unless user specify it
     const settings = Object.assign({ persistent: true, durable: true }, options);
-    let message = typeof msg === 'string' ? msg : Object.assign({}, msg);
+    let message = ((typeof msg === 'string') || Array.isArray(msg)) ? msg : Object.assign({}, msg);
```

Comment:
How about using Array.from(msg) or Object.assign([], msg) to protect input and be compliant to existing object handling?

Reply:
fixed